### ISSUE TITLE
 Understanding Flow SPTransform

### ISSUE THREAD

**Issue by YoussefFathi** [lliuz/ARFlow#4]
I would appreciate it if you can clarify for me the following points related to specific parts of the "sp_transforms.py" file : 

1. In  **_transform_flow()_** , what is the purpose of the following block of code ?

       `        # inverse transform coords
        x0, y0 = self.inverse_transform_coords(
            width=width, height=height, thetas=theta1)

        x1, y1 = self.inverse_transform_coords(
            width=width, height=height, thetas=theta2, offset_x=u, offset_y=v)

        # subtract and create new flow
        u = x1 - x0
        v = y1 - y0
        new_flow = torch.stack([u, v], dim=1)`

2. Why didn't you use only the following part of the code to transform the optical flow just like the _**transform_image()**_ function ?

         `  # transform coords
        xq, yq = self.transform_coords(width=width, height=height, thetas=theta1)

        # interp2
        transformed = self._flow_interp2(new_flow, xq, yq)`

3. What is the difference between the functionality of _**inverse_transform_coords()**_ and _**transform_coords()**_  ?

**Comment by lliuz** [comment]
Since spatial transformations will lead in a change to coordinates, we should sample optical flow on the view of transformed coordinates rather than the original one.

In this code, `inverse_transform_coords()`  transform coordinates:
![image](https://user-images.githubusercontent.com/20420011/83266226-218a6980-a1f5-11ea-85b6-20422476ab68.png)
and `transform_coords()` is its inverse operation.
